packages/eslint-plugin-react-hooks/__tests__/ESLintRuleExhaustiveDeps-test.js

```diff
@@ -5747,56 +5813,6 @@ const tests = {
         },
       ],
     },
-    {
-      code: normalizeIndent`
-        function MyComponent(props) {
-          let handleNext = () => {
-            console.log('hello');
-          };
-          if (props.foo) {
-            handleNext = () => {
-              console.log('hello');
-            };
-          }
-          useEffect(() => {
-            return Store.subscribe(handleNext);
-          }, [handleNext]);
-        }
-      `,
-      errors: [
-        {
-          message:
-            "The 'handleNext' function makes the dependencies of useEffect Hook " +
-            '(at line 13) change on every render. To fix this, wrap the ' +
-            "'handleNext' definition into its own useCallback() Hook.",
-          // Normally we'd suggest moving handleNext inside an
-          // effect. But it's used more than once.
-          // TODO: our autofix here isn't quite sufficient because
-          // it only wraps the first definition. But seems ok.
```

Comment:
I had decided at reassigned let/var values were too complex to reason about, so this case would no-longer be an error. How do we want to think about let/var declarations with reassignment?

I suppose we could say that if the initial assignment is a construction, then it's invalid, and if you later assign it another construction we won't try to catch that.

We could then safely suggest that (as in this case) you at least wrap your initial declaration in a useMemo/useCallback.

Reply:
>I suppose we could say that if the initial assignment is a construction, then it's invalid, and if you later assign it another construction we won't try to catch that.

Seems reasonable.

Reply:
This still needs to be changed?

Reply:
Done in https://github.com/facebook/react/pull/19590/commits/e0ca072197821d999012773faf329d181851a2f4